inventory/any.cf

```diff
@@ -196,6 +203,141 @@ bundle agent cfe_autorun_inventory_proc
       "$(this.bundle): we have kernel version '$(version)'";
 }
 
+bundle agent cfe_autorun_inventory_proc_cpuinfo
+{
+  classes:
+    "_have_cpuinfo" expression => isvariable("default:cfe_autorun_inventory_proc.cpuinfo_idx");
+
+      # So that we don't inventory non dereferenced variables we check to see
+      # if we have the info first This is only necessary because its currently
+      # invalid to do isvariable on an array key that contains a space
+      # Ref: redmine#7088 https://dev.cfengine.com/issues/7088
+      "have_cpuinfo_cpu_cores" expression => strcmp("cpu cores", "$(default:cfe_autorun_inventory_proc.cpuinfo_array[$(default:cfe_autorun_inventory_proc.cpuinfo_idx)][0])");
+      "have_cpuinfo_model_name" expression => strcmp("model name", "$(default:cfe_autorun_inventory_proc.cpuinfo_array[$(default:cfe_autorun_inventory_proc.cpuinfo_idx)][0])");
+
+      # Stubbed out hyperthreading, works but difficult to detect reliably? Limited value?
```

Comment:
I would not leave hyperthreading in a dangling state in comments. Just take it out, or make sure that it works 100% everywhere and take it in.